### PR TITLE
add a support matrix for backup, volumetypes, incremental

### PR DESCRIPTION
**Pull request by weshayutin**
## Why the changes were made

We need document and distinguish our support across backup types, volume types and incremental backup support.

## How to test the changes made

READ!

### CODE REVIEW

File: docs/supported_backups.md
Comment (shubham-pampattiwar) [inline docs/supported_backups.md:9]: Maybe also add another column of cloud provider snapshots ? (via VSL )

Comment (weshayutin) [reply docs/supported_backups.md:9]: ya.. I will before we merge. I just want to keep this clear atm for the doc team.
/hold

Comment (sseago) [reply docs/supported_backups.md:9]: @shubham-pampattiwar We'll need another dimension in the chart for that. Based on VolumeMode, CSI and native snapshots are supported in the same cases (i.e. all). We'll need to add columns for "Volume uses CSI driver" and "Cloud Volume (aws/gcp/azure)" -- so if the first new column is false, then CSI/DM are `N`, and if the second new column is false, then native snapshots are `N`

File: docs/supported_backups.md
Comment (shubham-pampattiwar) [inline docs/supported_backups.md:29]: `s/**Note/**Note**`

File: docs/supported_backups.md
Comment (weshayutin) [reply docs/supported_backups.md]: added a section for volume snapshots

File: docs/supported_backups.md
Comment (sseago) [inline docs/supported_backups.md:9]: Calling it "Volume Snapshot" is confusing, since VolumeSnapshot is the CR created for CSI snapshots. Maybe "native snapshots (VolumeSnapshotter plugin)? If that's too long, then just "Native snapshots" and a note below the chart saying that native snapshots are only supported if there's an included VolumeSnapshotter plugin.

Comment (sseago) [reply docs/supported_backups.md:9]: Oh, there's already a note below for the scope of this column. In that case, either "Native snapshots" or "Native snapshots (VolumeSnapshotter plugin)" for the heading here, and use the same wording beginning the note below.

Comment (shubham-pampattiwar) [reply docs/supported_backups.md:9]: Cloud Provider Native Snapshots

Comment (weshayutin) [reply docs/supported_backups.md:9]: nice, thank you both!

File: docs/supported_backups.md
Comment (weshayutin) [reply docs/supported_backups.md]: updated to:
  * ii -  DataMover upload and download operations use kopia regardless of uploader type.

**Comment by kaovilai** [comment]
Plugin Native Snapshots sounds clearer..

Both **Plugin Native Snapshots** and **CSI Snapshot** could trigger snapshot at the Cloud Provider natively.

File: docs/supported_backups.md
Comment (shubham-pampattiwar) [inline docs/supported_backups.md:29]: Maybe just update this to: Non CSI Volume support for AWS, GCP, Azure via VolumeSnapshotter plugins.
WDYT @kaovilai ?

Comment (kaovilai) [reply docs/supported_backups.md:29]: Sure. Perhaps also have to change description to say

```diff
+OADP supports volume snapshot operations for OpenShift Virtualization workloads.
-OADP supports backup and restore operations for OpenShift Virtualization workloads.
```

Comment (weshayutin) [reply docs/supported_backups.md:29]: I don't think `volume snapshot operations` is clear enough for customers.  They are using OADP to backup and resore or disaster recovery.  Not to snapshot volumes.

Comment (sseago) [inline docs/supported_backups.md:29]: non-CSI is inaccurate. gp3-csi, for example, also supports native snapshots.

Comment (weshayutin) [reply docs/supported_backups.md]: Thank you @sseago, updated and added links to plugins

**Review by kaovilai — APPROVED** [review]

**Review by sseago — APPROVED** [review]

**Pull request merged**
